### ISSUE TITLE
Error: Class 'Kevinrob\GuzzleCache\CacheMiddleware' not found

### ISSUE THREAD

**Issue by bradynpoulsen** [launchdarkly/php-client#65]
## Reproduction Steps

1. Install via Composer

```sh
$ composer require launchdarkly/launchdarkly-php
```

2. Construct an instance of the `LDClient`

```php
$client = new LaunchDarkly\LDClient('sdk-...');
```

## Expected Outcome

`LaunchDarkly\GuzzleFeatureRequester` will try to utilize the `Kevinrob\GuzzleCache\CacheMiddleware` class if the class is available

## Actual Outcome

```
Error: Class 'Kevinrob\GuzzleCache\CacheMiddleware' not found
```

## Proposed Resolution

Allow for utilization of the `GuzzleFeatureRequester` without requiring a 3rd party caching handler in scenarios that `Kevinrob\GuzzleCache\CacheMiddleware` is not suitable (ie, single run containers that discard their filesystem and memory after use)

 1. Modify `LaunchDarkly\GuzzleFeatureRequester` to utilize `Kevinrob\GuzzleCache\CacheMiddleware` if it is available via `class_exists`
 2. Add a message to the `kevinrob/guzzle-cache-middleware` composer.json suggestion describing its role in the library

**Comment by djbusby** [comment]
The documentation leads ones to believe that KevinRob is an optional dependency but in practice it must be manually added to composer.  It shows in the 'Using Guzzle' section.  Maybe what's happening is that if Guzzle if found (cause it's part of another dependency) then LD assumes KevinRob is available.

**Comment by bradynpoulsen** [comment]
The KevinRob cache is only in memory unless you configure the `LDClient`'s "cache" option, otherwise it's only a runtime in-memory cache. That is only useful if you are making multiple http calls though.. Perhaps another option is to only try to utilize KevinRob if the `cache` option is configured

**Comment by arun251** [comment]
@bradynpoulsen how do the changes in #78 look to you?

**Comment by arun251** [comment]
The changes were released in 2.1.2.
